test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,523 @@
+    }
+
+    private List<CompletableFuture<Void>> createEventReaders(int readers, ClientFactoryImpl factory, String readerGroupName) {
+        log.info("Creating {} Readers.", readers);
+        List<CompletableFuture<Void>> readerList = new ArrayList<>();
+        for (int i = 0; i < readers; i++) {
+            String readerId = String.format("LargeEventReader-%d", i);
+            readerList.add(startNewReader(readerId,
+                    factory,
+                    readerGroupName,
+                    eventsReadFromPravega,
+                    eventWriteCount,
+                    eventReadCount,
+                    stopReadFlag));
+        }
+        return readerList;
+    }
+
+    public void validateCleanUp(String streamName) throws ExecutionException, InterruptedException {
+        // Seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(SCOPE_NAME, streamName);
+        log.info("Sealing stream {}", streamName);
+        assertTrue(sealStreamStatus.get());
+        assertTrue(sealStreamStatus.get());
+
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(SCOPE_NAME, streamName);
+        log.info("Deleting stream '{}'", streamName);
+        assertTrue(deleteStreamStatus.get());
+        //delete the  SCOPE_NAME
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(SCOPE_NAME);
+        log.info("Deleting scope '{}'", SCOPE_NAME);
+        assertTrue(deleteScopeStatus.get());
+    }
+
+    @Test
+    public void readWriteTest() throws ExecutionException, InterruptedException {
+        String readerGroupName = "testLargeEventReaderGroup";
+        String streamName = "ReadWrite";
+        StreamConfiguration config = getStreamConfiguraton(NUM_READERS);
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, Serializer.MAX_EVENT_SIZE * 5);
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE_NAME, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createEventWriters(streamName, NUM_WRITERS, clientFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            ExecutorServiceHelpers.shutdown(writerPool);
+            stopReadFlag.set(true);
+            Futures.allOf(readers).get();
+            ExecutorServiceHelpers.shutdown(readerPool);
```

Comment:
You don't need to do this here. The `teardown` method will take care of it, won't it? Same with the tests below.